sphinx/domains/python.py

```diff
@@ -641,6 +641,152 @@ def generate(self, docnames=None):
         return sorted_content, collapse
 
 
+class PythonClassIndex(Index):
+    """
+    Index subclass to provide the Python class index.
+    """
+
+    name = 'classindex'
+    localname = l_('Python Class Index')
+    shortname = l_('classes')
+
+    def generate(self, docnames=None):
+        # type: (List[unicode]) -> Tuple[List[Tuple[unicode, List[List[Union[unicode, int]]]]], bool]  # NOQA
+        content = {}  # type: Dict[unicode, List]
+        # list of prefixes to ignore
+        ignores = None  # type: List[unicode]
+        ignores = self.domain.env.config['classindex_common_prefix']  # type: ignore
+        ignores = sorted(ignores, key=len, reverse=True)
+        # list of all classes, sorted by class name
+        unsorted_classes = [(refname, (docname, type))
+                            for (refname, (docname, type)) in iteritems(self.domain.data['objects'])
```

Comment:
In addition, Mr flake8 warns this line is too long. Please fold the line :-)
Line 735 is also.
https://travis-ci.org/sphinx-doc/sphinx/jobs/186467941#L441